contrib/vagrant/provision-util.sh

```diff
@@ -179,6 +185,156 @@ os::provision::get-network-plugin() {
   echo "${plugin}"
 }
 
+os::provision::install-networking() {
+  local plugin=$1
+  local master_ip=$2
+  local origin_root=$3
+  local config_root=$4
+  local is_master=${5:-false}
+
+  echo "Configuring networking"
+
+  if [[ "${plugin}" =~ redhat/ ]]; then
+    os::provision::install-sdn "${origin_root}"
+  elif [[ "${plugin}" = "flannel" ]]; then
+    os::provision::install-flannel "${master_ip}" "${origin_root}" \
+        "${config_root}" "${is_master}"
+  else
+    >&2 echo "Unable to deploy network plugin ${plugin}"
+    exit 1
+  fi
+}
+
+os::provision::install-flannel() {
+  local master_ip=$1
+  local origin_root=$2
+  local config_root=$3
+  local is_master=$4
+
+  if os::provision::in-container; then
+    local if_name=eth0
+  else
+    yum install -y flannel
+
+    local conf_path=/etc/sysconfig/network-scripts/
+    local if_to_edit=$(find ${conf_path}ifcfg-* | xargs grep -l VAGRANT-BEGIN)
+    local if_name=`echo ${if_to_edit} | awk -F- '{ print $3 }'`
+  fi
+
+  if [ "${is_master}" = "true" ]; then
+    os::provision::flannel-configure-master "${master_ip}" "${origin_root}" \
+       "${config_root}" "${if_name}"
+  else
+    local daemon_opts="-iface=${if_name} -ip-masq=false -v=5 \
+-remote=${master_ip}:8080"
+    cat <<EOF > /etc/sysconfig/flanneld
+FLANNEL_OPTIONS="${daemon_opts}"
+EOF
+
+    # Rewrite flanneld unit ExecStart to remove etcd arguments
+    sed -i.bak -e 's+\(ExecStart=/usr/bin/flanneld\)+\1 $FLANNEL_OPTIONS+' \
+        /lib/systemd/system/flanneld.service
```

Comment:
Please copy to /etc/systemd/system and edit it in /etc  (/lib is for distros, /etc is for local machines)